packages/data/src/components/use-dispatch/use-dispatch-with-map.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { mapValues } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useRegistry from '../registry-provider/use-registry';
+
+function proxyDispatch( propName, registry, ...args ) {
+	this( registry.dispatch, registry )[ propName ]( ...args );
+}
+
+const useDispatchWithMap = ( dispatchMap ) => {
+	const registry = useRegistry();
+	const currentDispatchProps = dispatchMap( registry.dispatch, registry );
```

Comment:
This is the cause of the performance losses. Inline this variable (in line 24) so it only runs when the memo cache is cleared instead of on every render.

Reply:
🤦‍♂ of course!

Reply:
fixed in 7d88dd1.  A little improvement in performance but not that much.  likely the props change too much in withDispatch to cause a noticeable difference.